lib/haml_lint/linter/class_attribute_with_static_value.rb

```diff
@@ -9,11 +9,17 @@ module HamlLint
   # ...over:
   #
   #   %tag{ class: 'class-name' }
+  #
+  # But will allow invalid class names for templating:
+  #
+  #   %tag{ class: '{{ template-var }}' }
   class Linter::ClassAttributeWithStaticValue < Linter
     include LinterRegistry
 
     STATIC_TYPES = [:str, :sym].freeze
 
+    VALID_CLASS_REGEX = /^-?[_a-zA-Z]+[_a-zA-Z0-9-]*$/
```

Comment:
@trotzig I thought the same! The officially spec'd regex is much more complex, and it should be, but you're right the number of edge cases are significantly limited.